### DISCUSSION TITLE
Regarding ALGORITHMS

### DISCUSSION THREAD

**Discussion by PoojaGaur1708** [DMTF/libspdm#844] [Q&A] [answered]
Hello , Algorithm fails at below step in libspdm_try_negotiate_algorithms L383.
Can you pls suggest !

 if (libspdm_is_capabilities_flag_supported(
377                 spdm_context, true,
378                 SPDM_GET_CAPABILITIES_REQUEST_FLAGS_KEY_EX_CAP,
379                 SPDM_GET_CAPABILITIES_RESPONSE_FLAGS_KEY_EX_CAP)) {
380             algo_size = libspdm_get_dhe_pub_key_size(
381                 spdm_context->connection_info.algorithm
382                 .dhe_named_group);
383             if (algo_size == 0) {
384         printf("%s:%d***\n",__func__,__LINE__);
385                 status = LIBSPDM_STATUS_NEGOTIATION_FAIL;
386                 goto receive_done;
387             }

Get Capability 0000: 11 e1 00 00 00 00 00 00 c0 c2 00 00
Capability :0000: 11 61 00 00 00 00 00 00 c0 c2 00 00 

Negotiate Algorithm Requester sending : 
0000: 11 e3 04 00 30 00 00 00 00 00 00 00 02 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 
0020: 02 20 00 00 03 20 04 00 04 20 00 00 05 20 01 00

Algorithm Responder sending :
0000: 11 63 04 00 34 00 00 00 02 00 00 00 00 00 00 00 02 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 
0020: 00 00 00 00 02 20 00 00 03 20 04 00 04 20 00 00 05 20 01 00 00

**Comment by steven-bellock** [comment, marked as answer]
It looks both the Requester and Responder support KEY_EXCHANGE, however the Requester did not specify any DHE algorithms like `secp256r1`. We can probably put an assertion in the code to catch this.

**Reply by PoojaGaur1708** [reply]
yes resolved . thanks.
